mealplanner-ui/src/pages/Meals/Meals.tsx

```diff
@@ -26,59 +29,101 @@ const mealsQuery = graphql`
     { fetchPolicy: "store-or-network" }
   );
+
+  const selectedTags = data.gqLocalState.selectedMealTags || [];
+
   return (
+    <div>
     <Grid
       container
       spacing={2}
       columns={2}
-      justifyContent="center"
-      marginTop="1rem"
+      gap="2rem"
+      margin="1rem 2rem"
+      width="95%"
+      justifyContent="space-between"
     >
-      <Paper
-        component="form"
-        sx={{
-          display: "flex",
-          alignItems: "center",
-          width: "75%",
-          justifyContent: "center",
+      <FormControl component="fieldset">
+      <RadioGroup
+        row
+        aria-label="searchType"
+        name="searchType"
+        value={searchType}
+        onChange={(e) => {
+          setSearchType(e.target.value);
         }}
       >
-        <InputBase
-          sx={{ ml: 1, flex: 1 }}
-          placeholder="Search Meal"
-          inputProps={{ "aria-label": "Search Meal" }}
-          onChange={(e) => setSearchMeal(e.target.value.toLowerCase())}
+       <FormControlLabel
+          value="name"
+          control={
+            <Radio 
+              checked={searchType === 'name'}
+            />
+          }
+          label={
+            <InputBase
+              placeholder="Search Meal"
+              inputProps={{ "aria-label": "Search Meal" }}
+              readOnly={searchType !== 'name'}
+              value={searchMeal}
+              onChange={(e) => {
+                if (searchType === 'name') {
+                  setSearchMeal(e.target.value.toLowerCase());
+                }
+              }}
+              style={{ cursor: 'text',
+                      borderBottom: '1px solid black', width: '40vw' }}
+              />
+          }
+        />
+        <FormControlLabel
+          value="favorites"
+          control={<Radio />}
+          label="Favorites"
+          checked={searchType === 'favorites'}
         />
-        <Search />
-      </Paper>
-      {data.meals ? (
-        <Grid
-          container
-          spacing={2}
-          justifyContent="center"
-          marginTop="1rem"
-          columns={4}
-        >
-          {data.meals?.nodes.map((node) => {
-            if (node.nameEn.toLowerCase().includes(searchMeal))
-              return <MealCard node={node} />;
-          })}
-        </Grid>
+        <FormControlLabel
+          value="tags"
+          control={<Radio />}
+          label="Tags"
+          checked={searchType === 'tags'}
+        /> 
+      </RadioGroup>
+    </FormControl>
+    </Grid>
+      {searchType === 'tags' &&
+        <Suspense fallback={"loading tags..."}>
+        <MealTags tags={data}/>
+      </Suspense>
+    }
+   
+    {data.meals ? (
+      <Grid container spacing={2} margin="1rem" columns={4}>
+        {data.meals?.nodes.map((node) => {
+      if ((searchType === 'name' && node.nameEn.toLowerCase().includes(searchMeal)) || (searchType === 'tags' && selectedTags.every(tag => node.tags?.includes(tag)))) {
```

Comment:
Remove extra parentheses around && operation. It has higher priority vs || so the parentheses are not required.

Reply:
It is for clarity that the extra paranthesis is added, basically better readability of code.